hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -207,9 +206,21 @@ chainSyncHandler tracer callback headState =
 
   onRollForward :: Block -> m ()
   onRollForward blk = do
+    let point = blockPoint blk
+    traceWith tracer $ RolledForward $ SomePoint point
+
+    let chainPoint = fromConsensusPointHF point
+        slotNo = case chainPoint of
+          ChainPointAtGenesis -> 0
+          ChainPoint s _ -> s
+    case slotToUTCTime timeHandle slotNo of
+      Left reason ->
+        traceWith tracer $ TickTimeConversionFailed{slotNo, reason}
```

Comment:
This ought to raise an exception, not a log message. If we ever hit that case, it probably means that the application will be totally unusable from then on. I mean, we have just received a block from the node for which we can't calculate the current time? That is an exceptional case and truth is, we don't know how to handle that.

Reply:
That's correct! I will convert this to an exception.

Reply:
Done